File: app/src/main/java/hcm/ditagis/com/tanhoa/qlsc/utities/MyServiceFeatureTable.java
package hcm.ditagis.com.tanhoa.qlsc.utities;

import com.esri.arcgisruntime.data.ServiceFeatureTable;
import com.esri.arcgisruntime.layers.ArcGISMapImageLayer;
import com.esri.arcgisruntime.layers.ArcGISMapImageSublayer;
import com.esri.arcgisruntime.loadable.LoadStatus;

public class MyServiceFeatureTable {
    private ServiceFeatureTable SFTLayerHanhChinh;

    private MyServiceFeatureTable(final ArcGISMapImageLayer arcGISMapImageLayer) {
        final ArcGISMapImageSublayer layerHanhChinh = (ArcGISMapImageSublayer) arcGISMapImageLayer.getSublayers().get(3);
        layerHanhChinh.loadAsync();
        layerHanhChinh.addDoneLoadingListener(new Runnable() {
            @Override
            public void run() {
                if (layerHanhChinh.getLoadStatus() == LoadStatus.LOADED) {
                    SFTLayerHanhChinh = layerHanhChinh.getTable();
                }
            }
        });


    }

    private static MyServiceFeatureTable instance = null;

    public static MyServiceFeatureTable getInstance(ArcGISMapImageLayer arcGISMapImageLayer) {
        if (instance == null) {
            instance = new MyServiceFeatureTable(arcGISMapImageLayer);
        }
        return instance;
    }


    public ServiceFeatureTable getSFTLayerHanhChinh() {
        return SFTLayerHanhChinh;
    }
}

File: app/src/main/java/vn/ditagis/com/tanhoa/qlsc/utities/MyServiceFeatureTable.kt
package vn.ditagis.com.tanhoa.qlsc.utities

import com.esri.arcgisruntime.data.ServiceFeatureTable
import com.esri.arcgisruntime.layers.ArcGISMapImageLayer
import com.esri.arcgisruntime.layers.ArcGISMapImageSublayer
import com.esri.arcgisruntime.loadable.LoadStatus

class MyServiceFeatureTable private constructor(arcGISMapImageLayer: ArcGISMapImageLayer) {
    var sftLayerHanhChinh: ServiceFeatureTable? = null
        private set

    init {
        val layerHanhChinh = arcGISMapImageLayer.sublayers[3] as ArcGISMapImageSublayer
        layerHanhChinh.loadAsync()
        layerHanhChinh.addDoneLoadingListener {
            if (layerHanhChinh.loadStatus == LoadStatus.LOADED) {
                sftLayerHanhChinh = layerHanhChinh.table
            }
        }


    }

    companion object {

        var instance: MyServiceFeatureTable? = null

        fun getInstance(arcGISMapImageLayer: ArcGISMapImageLayer): MyServiceFeatureTable? {
            if (instance == null) {
                instance = MyServiceFeatureTable(arcGISMapImageLayer)
            }
            return instance
        }
    }
}
